ingestion/src/metadata/ingestion/models/custom_pydantic.py

```diff
@@ -30,13 +32,187 @@
 SECRET = "secret:"
 JSON_ENCODERS = "json_encoders"
 
+RESTRICTED_KEYWORDS = ["::", ">"]
+RESERVED_COLON_KEYWORD = "__reserved__colon__"
```

Comment:
What happens if a table or column has this identifier in the name? If we are not planning to support such a scheme its best to fail for entities that have these strings in the original names.